File: algorithms/code/leetcode/lc009_palindrome_number/LC009PalindromeNumberTests.java
package leetcode.lc009_palindrome_number;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public final class LC009PalindromeNumberTests {
    @Test
    public void test213() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertEquals(false, solution.isPalindrome(213));
    }

    @Test
    public void test456() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertEquals(false, solution.isPalindrome(456));
    }

    @Test
    public void test454() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertEquals(true, solution.isPalindrome(454));
    }

    @Test
    public void test99() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertEquals(true, solution.isPalindrome(99));
    }

    @Test
    public void test1() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertEquals(true, solution.isPalindrome(1));
    }

    @Test
    public void test10() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertEquals(false, solution.isPalindrome(10));
    }

    @Test
    public void testMinus1() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertFalse(solution.isPalindrome(-1));
    }

    @Test
    public void test0() throws Exception {
        LC009PalindromeNumber solution;
        solution = new LC009PalindromeNumber();
        assertTrue(solution.isPalindrome(0));
    }
}

File: algorithms/code/leetcode/lc009_palindrome_number/LC009PalindromeNumber.Tests.cs
using Xunit;

namespace AlgorithmDesign.code.leetcode.lc009_palindrome_number
{
    public class SolutionTests
    {
        [Fact]
        public void Test213()
        {
            Assert.False(new Solution().IsPalindrome(213));
        }

        [Fact]
        public void Test456()
        {
            Assert.False(new Solution().IsPalindrome(456));
        }

        [Fact]
        public void Test454()
        {
            Assert.True(new Solution().IsPalindrome(454));
        }

        [Fact]
        public void Test99()
        {
            Assert.True(new Solution().IsPalindrome(99));
        }

        [Fact]
        public void Test1()
        {
            Assert.True(new Solution().IsPalindrome(1));
        }

        [Fact]
        public void Test10()
        {
            Assert.False(new Solution().IsPalindrome(10));
        }

        [Fact]
        public void TestMinus1()
        {
            Assert.False(new Solution().IsPalindrome(-1));
        }

        [Fact]
        public void Test0()
        {
            Assert.True(new Solution().IsPalindrome(0));
        }
    }
}
